prodsys/simulation/resources.py

```diff
@@ -206,11 +222,16 @@ def get_processes(self, process: process.PROCESS_UNION) -> List[state.State]:
         Returns:
             List[state.State]: The state of the resource for the process.
         """
-        possible_states = [
-            actual_state
-            for actual_state in self.production_states
-            if actual_state.state_data.ID == process.process_data.ID
-        ]
+        from prodsys.simulation.process import RequiredCapabilityProcess
+        possible_states = []
+        for actual_state in self.production_states:
+            if isinstance(process, RequiredCapabilityProcess) and (
```

Comment:
the capability of the requiredCapabilityProcess should be used, as for all other processes, in the match_request function. Here is logic for process matching -> move it to the process class.

Reply:
Commit 29: But here it is no match with a request, but from a resource it s capability. I added the match_request in the process.